File: src/main/java/com/kursovaya/entity/Order.java
package com.kursovaya.entity;

import com.kursovaya.entity.order.OrderDetails;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "orders")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Order implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "order_id", unique = true, nullable = false)
    Long id;

    @OneToMany(mappedBy="order", cascade = CascadeType.ALL)
    private List<OrderDetails> orderDetails = new ArrayList<>();
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    User user;

    private String orderStatus;
}

File: supportportalapp/src/app/model/Order/order.ts
import {Product} from "../Product/product";
import {User} from "../user";
import {ProductInOrder} from "./productInOrder";

export class Order {

  constructor(
    public id: number = null,
    public orderDetails: ProductInOrder[] = null,
    public user: User = null,
    public orderStatus: string= null,
  ) {}

}
